Comments/ClientApp/src/components/GenerateCode/__tests__/GenerateCode.test.js

```diff
@@ -1,56 +1,48 @@
-/* eslint-env jest */
 import React from "react";
-import { mount, shallow } from "enzyme";
-import toJson from "enzyme-to-json";
-import { Button } from "@nice-digital/nds-button";
-
+import { render, screen, fireEvent } from "@testing-library/react";
 import { GenerateCode } from "../GenerateCode";
 
-describe("GenerateCode", () => {
-
-	const fakePropsNoCode = {
-		organisationCodes: [{
-			canGenerateCollationCode: true,
-			collationCode: null,
-			organisationAuthorisationId: 0,
-			organisationId: 999,
-			organisationName: "Super Magic Club",
-		}],
-		consultationId: 111,
-	};
+const fakePropsNoCode = {
+	organisationCodes: [{
+		canGenerateCollationCode: true,
+		collationCode: null,
+		organisationAuthorisationId: 0,
+		organisationId: 999,
+		organisationName: "Super Magic Club",
+	}],
+	consultationId: 111,
+};
+
+test("should match snapshot with no code having been generated", () => {
+	const {container} = render(<GenerateCode {...fakePropsNoCode} />);
+	const shareOrganisationButton = screen.getByRole("button", { name: "Share with organisation" });
+	fireEvent.click(shareOrganisationButton);
+	expect(container).toMatchSnapshot();
+});
 
+test("should match snapshot with code having been generated", () => {
 	let fakePropsCode = {
 		organisationCodes: [{...fakePropsNoCode.organisationCodes[0]}],
```

Comment:
is this testing for collationCode null or is it meant to be populated? Not quite following :-)

Reply:
It's snapshot testing for when a code has been generated. We're copying parts of the fakePropsNoCode object and then setting all the other code properties.